extensions/cms/src/cmsResource/cmsResourceCommands.ts

```diff
@@ -76,16 +73,16 @@ export function addRegisteredServerCommand(appContext: AppContext, tree: CmsReso
 		let relativePath = node instanceof CmsResourceTreeNode ? '' : node.relativePath;
 		let serverName = node instanceof CmsResourceTreeNode ? node.connection.options.registeredServerName === ''
 			? node.connection.options.server : node.connection.options.registeredServerName : null;
-		await appContext.cmsUtils.addRegisteredServer(relativePath, node.ownerUri, serverName).then((result) => {
-			if (result) {
-				tree.notifyNodeChanged(node);
-			}
-		}, (error) => {
+		try {
+			await appContext.cmsUtils.addRegisteredServer(relativePath, node.ownerUri, serverName);
+			tree.notifyNodeChanged(node);
+		} catch (error) {
 			// error out
-			let errorText = localize('cms.errors.addRegisterServerFail', 'Could not add the Registered Server {0}', error);
+			let errorText = error ? error.message : localize('cms.errors.addRegisterServerFail', 'Could not add the Registered Server {0}', error);
 			appContext.apiWrapper.showErrorMessage(errorText);
-			return;
-		});
+			error = error ? error : new Error(errorText);
+			throw error;
```

Comment:
Error can't be undefined here so this check is pointless (and the error text will always just be the message)